Donor/Repositories/DonorRepository.cs

```diff
@@ -1,80 +1,105 @@
 ﻿using Donor.DbContexts;
+using Donor.Entities;
 using Microsoft.Data.SqlClient;
 using Microsoft.EntityFrameworkCore;
+using Microsoft.Extensions.Logging;
 using Serilog;
+using System;
 using System.Collections.Generic;
+using System.ComponentModel.DataAnnotations;
 using System.Threading.Tasks;
+using System.Linq;
 
 namespace Donor.Repositories
 {
     public class DonorRepository : IDonorRepository
     {
         private readonly DonorContext _context;
-		private readonly ILogger<DonorRepository> _log;
+        private readonly ILogger<DonorRepository> _log;
 
-		public DonorRepository(DonorContext context, ILogger<DonorRepository> log)
+        public DonorRepository(DonorContext context, ILogger<DonorRepository> log)
         {
             _context = context ?? throw new ArgumentNullException(nameof(context));
-			_log = log ?? throw new ArgumentNullException(nameof(log));
-		}
-	
+            _log = log ?? throw new ArgumentNullException(nameof(log));
+        }
 
-        public async Task AddDonorAsync(Entities.Donor donor)
+        public async Task AddDonorAsync(Entities.Donor donor, List<int> donationPreferences)
         {
-			try
-			{
-				await _context.Donors.AddAsync(donor);				
+            using var transaction = await _context.Database.BeginTransactionAsync();
+            try
+            {
+                await _context.Donors.AddAsync(donor);
+                await _context.SaveChangesAsync();
```

Comment:
We should try to make one roundtrip to the DB per operation.
This is One roundtrip to add the Donor which is good !